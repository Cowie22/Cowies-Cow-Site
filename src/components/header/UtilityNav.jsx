import React from 'react'
import { Container, Row, Col } from 'react-bootstrap'

const UtilityNav = (props) => {
  return (
    <section className='utility-nav'>
      <Container>
        <Row>
          <Col xl={{ span: 12, offset: 0 }} lg={{ span: 12, offset: 0 }}>
            <nav className='utility-nav-container'>
              <Row>
                <Col xl={{span: 9, offset: 3}} lg={{span: 10, offset: 2}}>
                  <ul className='utility-left-container'>
                    <li>
                      <a
                        className='utility-header-nav'
                        href='#isi'
                      >
                        Important Safety Information
                      </a>
                    </li>
                    <li className='center-nav'>
                      <a
                        target='_blank'
                        rel='noopener noreferrer'
                        className='utility-header-nav'
                        href='https://www.myovant.com/myfembree-prescribing-information.pdf'
                        data-testid='Prescribing Information'
                      >
                        <span>Prescribing Information, including <b>BOXED WARNING</b></span>
                      </a>
                    </li>
                    <li>
                      <a
                        target='_blank'
                        rel='noopener noreferrer'
                        className='utility-header-nav'
                        href='https://www.Myfembree.com/'
                      >
                        Visit Patient Site
                      </a>
                    </li>
                  </ul>
                </Col>
              </Row>
            </nav>
          </Col>
        </Row>
      </Container>
    </section>
  )
}

export default UtilityNav