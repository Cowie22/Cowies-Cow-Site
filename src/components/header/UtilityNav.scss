.utility-nav {
  background: #F1F7F4;
  .utility-nav-container {
    width: 100%;
    display: inline;
    .utility-left-container {
      display: flex;
      justify-content: space-between;
      li {
        a {
          transition: all .5s;
          transition-delay: .2s;
          color: $grey;
        }
        &:hover {
          a {
            transition: all .5s;
            transition-delay: .2s;
            color: $purple;
          }
        }
      }
    }
  }
  p {
    font-size: 12px;
  }
  li {
    color: $black;
    cursor: pointer;
    font-size: 12px;
    &:visited {
      text-decoration: none;
    }
    &:active {
      text-decoration: none;
    }
  }
  .utility-header-nav {
    color: #333333;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    padding: 8px 4px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-family: $sans;
  }
  .utility-nav-end-container {
    background: #E3EFE9;
    text-align: center;
    font-family: $sans;
    width: 100%;
    height: 100%;
    padding: 8px 24px;
    ul {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 100%;
    }
  }
  .utility-nav-end-container li {
    color: #333333;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    font-family: $sans;
    a {
      width: 100%;
      transition: .5s all;
      transition-delay: .2s;
    }
    &:hover {
      cursor: pointer;
      a {
        transition: .5s all;
        transition-delay: .2s;
        color: $purple;
      }
    }
  }
}

@media screen and (max-width: 992px)  {
  .utility-nav {
    padding: 0;
    margin: 0;
    display: flex;
    justify-content: space-between;
    align-items: center;
    .utility-header-nav {
      font-size: 12px;
      line-height: 15px;
    }
    .utility-nav-container {
      .utility-left-container {
        justify-content: space-between;
        li {
          width: 33%;
          padding: 0px 4px;
          border-right: 1px solid $white;
          &:last-of-type {
            border: none;
          }
        }
      }
      .utility-right-col {
        padding: 0;
      }
      li {
        text-align: center;
        width: 100%;
        font-size: 12px;
        line-height: 15px;
      }
    }
  }
}