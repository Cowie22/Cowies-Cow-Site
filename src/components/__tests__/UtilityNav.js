import React from 'react'
import { render } from '@testing-library/react'
import UtilityNav from '../header/UtilityNav'

const { debug, getByAltText, getByText, getByTestId, getAllByText, container } = render( <UtilityNav /> )

const importantSafetyInformation = getByText('Important Safety Information')
const prescribingInformation = getByTestId('Prescribing Information')
const patientSite = getByText('Visit Patient Site')

test('Utility Nav links exist', () => {
  // debug()
  expect(importantSafetyInformation).toBeTruthy()
  expect(prescribingInformation).toBeTruthy()
  expect(patientSite).toBeTruthy()
})

test('Utility Nav links have appropriate href', () => {
  expect(importantSafetyInformation.getAttribute('href')).toBe('#isi')
  expect(prescribingInformation.getAttribute('href')).toBe('https://www.myovant.com/myfembree-prescribing-information.pdf')
  expect(patientSite.getAttribute('href')).toBe('https://www.Myfembree.com/')
})

test('Utility Nav snapshot matches', () => {
//   return
  // once the component is done, use snapshot
  expect(container.firstChild).toMatchSnapshot()
})