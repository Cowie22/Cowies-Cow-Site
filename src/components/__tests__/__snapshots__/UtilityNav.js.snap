// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utility Nav snapshot matches 1`] = `null`;
